base/sort.jl

```diff
@@ -407,6 +408,168 @@ julia> searchsortedlast([1, 2, 4, 5, 5, 7], 0) # no match, insert at start
 ```
 """ searchsortedlast
 
+function _insortedfirst(v::AbstractVector, x, lo::T, hi::T, o::Ordering)::Bool where T<:Integer
+    if lt(o, v[hi], x)
+        return false
+    end
+    u = T(1)
+    lo = lo - u
+    hi = hi + u
+    @inbounds while lo < hi - u
+        m = midpoint(lo, hi)
+        if lt(o, v[m], x)
+            lo = m
+        else
+            hi = m
+        end
+    end
+    if v[hi] == x
+        return true
+    end
+    return false
+end
+
+function _insortedlast(v::AbstractVector, x, lo::T, hi::T, o::Ordering)::Bool where T<:Integer
+    if lt(o, x, v[lo])
+        return false
+    end
+    u = T(1)
+    lo = lo - u
+    hi = hi + u
+    @inbounds while lo < hi - u
+        m = midpoint(lo, hi)
+        if lt(o, x, v[m])
+            hi = m
+        else
+            lo = m
+        end
+    end
+    if v[lo] == x
+        return true
+    end
+    return false
+end
+
+function insorted(v::AbstractVector, x, ilo::T, ihi::T, o::Ordering)::Bool where T<:Integer
```

Comment:
Do we want `in(x, coll)`, so `insorted(x, coll)` would be more intuitive argument order for me.

Reply:
Sure, I based my choice on `searchsorted` but it would be more logical to use the same calls as for `in`.